extensions/observability-devservices/testlibs/devresource-common/src/main/java/io/quarkus/observability/devresource/DevResourceLifecycleManager.java

```diff
@@ -28,8 +28,18 @@ public interface DevResourceLifecycleManager<T extends ContainerConfig> extends
      * @param configuration main observability configuration
      * @return module's config
      */
+    @Deprecated
     T config(ModulesConfiguration configuration);
 
+    /**
+     * Get resource's config from main observability configuration and extension catalog
+     *
+     * @param configuration main observability configuration
+     * @param catalog observability catalog. If OpenTelemetry or Micrometer are enabled.
+     * @return module's config
+     */
+    T config(ModulesConfiguration configuration, ExtensionsCatalog catalog);
```

Comment:
Have this one call deprecated by default -- so we don't break existing impls (if there are any ...).

Reply:
Not this one -- deprecate the old one.
And have the new one call the old one by default.

Reply:
I thought about it, but they are needed for different things and I didn't want to mess things too much.

Reply:
True ... but this way we can don't break any existing one (if there are some :-) ...